src/time-context.js

```diff
@@ -0,0 +1,81 @@
+const MINUTE_IN_MILLIS = 60 * 1000;
+
+const MINUTES = [
+    {key: ['time', 'minute'], value: 'getMinutes'},
+    {key: ['utc', 'time', 'minute'], value: 'getUTCMinutes'}
+];
+const HOURS = [
+    {key: ['time', 'hour'], value: 'getHours'},
+    {key: ['utc', 'time', 'hour'], value: 'getUTCHours'},
+    {key: ['hourOfDay'], value: 'getHours'},
+    {key: ['utc', 'hourOfDay'], value: 'getUTCHours'}
+];
+const DAYS = [
+    {key: ['date', 'day'], value: 'getDate'},
+    {key: ['utc', 'date', 'day'], value: 'getUTCDate'},
+    {key: ['dayOfWeek'], value: 'getDay'},
+    {key: ['utc', 'dayOfWeek'], value: 'getUTCDay'}
+];
+const MONTHS = [
+    {key: ['date', 'month'], value: 'getMonth', transform: function(v) { return v+1; }},
+    {key: ['utc', 'date', 'month'], value: 'getUTCMonth', transform: function(v) { return v+1; }}
+];
+const YEARS = [
+    {key: ['date', 'year'], value: 'getFullYear'},
+    {key: ['utc', 'date', 'year'], value: 'getUTCFullYear'}
+];
+
+const UNITS = [MINUTES, HOURS, DAYS, MONTHS, YEARS];
+
+
+function setValue(obj, date, depth, unit) {
+    const k = unit.key[depth];
+    if (depth === unit.key.length - 1) {
+      obj[k] = unit.transform ? unit.transform(date[unit.value]()) : date[unit.value]();
+      return;
+    }
+
+    let v = obj[k];
+    if (!v) {
+      v = {};
+      obj[k] = v;
+    }
+
+    setValue(v, date, depth+1, unit);
+}
+
+export function addDateTimeToContext(context, continueTimeUpdate) {
```

Comment:
Can you add a JSDoc comment to this function?

Reply:
Done